include/projfs.h

```diff
@@ -46,7 +46,7 @@ struct projfs;
 struct projfs_event {
 	struct projfs *fs;
 	uint64_t mask;			/* type flags; see projfs_notify.h */
-	pid_t pid;
+	int pid;
```

Comment:
Is there any way to hold onto using `pid_t` here?  It's definitely the right type from a C library or API perspective.

If the issue is how to define it correctly in the equivalent C# [`struct Event`](https://github.com/github/VFSForGit/blob/vfs-api/ProjFS.Linux/PrjFSLib.Linux.Managed/Interop/ProjFS.cs#L92), I notice that the Mono `Syscall` class just seems to map `pid_t` to `int` but adds an attribute when they define a C# version of [`struct Flock`](https://github.com/mono/mono/blob/master/mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs#L948).  I can't tell on a quick-ish read of the Mono repo, but I don't think they are translating this anywhere (e.g., via some clever wrapper).

I suppose we could also get the size of `pid_t` in the VFSForGit build at build-time, but then how to inject that into the C# code, I'm not sure OTTOMH.

Reply:
From a C library perspective, certainly, but I worry about a struct designed explicitly for outside consumption -- including via interop -- using types with non-obvious sizes. I don't think we could inject the size into the C# code -- there are no type aliases in C#, so we'd have to rewrite actual (human-editable) code files dynamically.

The attributes do indeed get used by [`create-native-map`](http://manpages.ubuntu.com/manpages/trusty/man1/create-native-map.1.html) -- see https://github.com/mono/mono-tools/tree/master/create-native-map. I haven't looked too deeply into how this works, but we can try using `pid_t` here again and see if it breaks later.